encoding/constructor_temp.go

```diff
@@ -0,0 +1,117 @@
+// Temporary file to hold constructors to ease merge conflicts.
+
+package encoding
+
+import "io"
+
+// STRING
+
+type stringEncoder struct{}
+
+// NewStringEncoder returns a new StringEncoder instance.
+func NewStringEncoder() StringEncoder {
+	return &stringEncoder{}
+}
+
+func (s *stringEncoder) Encode(values StringIterator, writer io.Writer) error {
```

Comment:
The writer should be moved to the ctor.

Reply:
Actually I think this is fine after giving it a second thought, although I'd probably move the writer to the first param though.